YugaByte/yugabyte-db
# Issue #1367 expose number of dead/unresponsive tservers as yb-master metric

The yb-master leader tracks yb-tserver heartbeats and knows which TServers haven't been able to heartbeat/have been unresponsive. The yb-master process exposes (via http://<master-ip>:7000/prometheus-metrics) a num_tablet_servers_live metric, but a num_tablet_servers_dead
might be more useful/easier for setting up alerts. This number being greater than 0 could be set to result in an alert.
    
To do this, expose another metric num_tablet_servers_dead from the catalog manager similar to the num_tablet_servers_live. This metric tracks the number of total tablet servers - live tablet servers.
    
Test plan:
1) Set up local yugabyte cluster with 3 nodes with prometheus monitoring
2) Verify that num_tablet_servers_live = 3 and num_tablet_servers_dead=0
on the :7000 prometheus metrics page
3) Use yb-ctl commands to stop one of the nodes
4) Verify that num_tablet_servers_live=2 and num_tablet_servers_dead=1
